cmd/gateway-azure.go

```diff
@@ -550,10 +552,25 @@ func (a *azureObjects) PutObjectPart(bucket, object, uploadID string, partID int
 		teeReader = io.TeeReader(data, io.MultiWriter(writers...))
 	}
 
-	id := azureGetBlockID(partID, etag)
-	err = a.client.PutBlockWithLength(bucket, object, id, uint64(size), teeReader, nil)
-	if err != nil {
-		return info, azureToObjectError(traceError(err), bucket, object)
+	subPartCount := size / azureBlockSize
+	lastSubPartSize := size - subPartCount*azureBlockSize
+	if lastSubPartSize != 0 {
+		subPartCount++
+	} else {
+		lastSubPartSize = azureBlockSize
+	}
+
```

Comment:
instead of pre-calculating subPartCount and lastSubPartSize above, doing it the following way looks easier to read (subjective opinion):
```
        remaining := size                                                                               
        subPartNumber := 1                                                                              
        for {                                                                                           
                subPartSize := remaining                                                                
                if subPartSize > azureBlockSize {                                                       
                        subPartSize = azureBlockSize                                                    
                }                                                                                       
                remaining = remaining - subPartSize                                                     
                id := azureGetBlockID(partID, subPartNumber, etag)                                      
                err = a.client.PutBlockWithLength(bucket, object, id, uint64(subPartSize), teeReader, n\
il)                                                                                                     
                if err != nil {                                                                         
                        return info, azureToObjectError(traceError(err), bucket, object)                
                }                                                                                       
                if remaining == 0 {                                                                     
                        break                                                                           
                }                                                                                       
                subPartNumber++                                                                         
        }
```

Reply:
Done